### ISSUE TITLE
Please improve pylance's parameter prompt layout style and provide function documentation

### ISSUE THREAD

**Issue by CatNofishing** [microsoft/pylance-release#301]
Compared with jedi, pylance's parameter prompt typesetting is too dense and not suitable for reading. In addition, for the same function, pylance only has parameter prompts, but no function documentation.
![image](https://user-images.githubusercontent.com/33288114/91679289-78983580-eb7a-11ea-84c0-ab97e55b392b.png)
![image](https://user-images.githubusercontent.com/33288114/91679641-8601ef80-eb7b-11ea-9e6d-70340405d295.png)

**Comment by CatNofishing** [comment]
@jakebailey @erictraut @mattetti

**Comment by jakebailey** [comment]
I had thought the original issue here would be fixed by the fixes for type aliases over-expanding, but it appears not:

![image](https://user-images.githubusercontent.com/5341706/95365692-55525b80-0887-11eb-95fd-00160e02413a.png)

Even though this is defined as:

```python
def scatter(
    x: ArrayLike,
    y: ArrayLike,
    s: Optional[Union[Scalar, ArrayLike]] = ...,
    c: Optional[Union[_ColorLike, Sequence[float], Sequence[_ColorLike]]] = ...,
    marker: Optional[MarkerStyle] = ...,
    cmap: Optional[Colormap] = ...,
    norm: Optional[Normalize] = ...,
    vmin: Optional[Scalar] = ...,
    vmax: Optional[Scalar] = ...,
    alpha: Optional[Scalar] = ...,
    linewidths: Optional[Union[Scalar, ArrayLike]] = ...,
    verts: Optional[Any] = ..., # not documented?
    edgecolors: Optional[Union[Literal["face", "none"], _ColorLike, Sequence[_ColorLike]]] = ...,
    *,
    plotnonfinite: bool = ...,
    data: Optional[Any] = ...,
    **kwargs: Any
) -> PathCollection: ...
```

And should be displayed much smaller thanks to the aliases.

**Comment by erictraut** [comment]
Aliases are still expanded when used within unions because the resulting union. I don't see any straightforward way to avoid this. Typescript also expands type aliases for unions.

**Comment by jakebailey** [comment]
I can understand within a union (unfortunate), but ArrayLike is an alias itself and its name is being elided for the `x` and `y` parameters.

**Comment by erictraut** [comment]
Yes, you're correct @jakebailey. There are places where the the type checker was transforming types unnecessarily and removing the type alias information in the process. I've checked in a fix, and the `ArrayLike` alias is now preserved. I don't know if this change will make it into this week's Pylance build. It might be too late. If not, it will be in the following release.

**Comment by jakebailey** [comment]
I'll try to sneak it in. The fix saves a few lines, though the nested unions still get expanded (as you said).

![image](https://user-images.githubusercontent.com/5341706/95386057-9eb0a400-08a3-11eb-8c87-7f1aee308966.png)

![image](https://user-images.githubusercontent.com/5341706/95386071-a112fe00-08a3-11eb-944a-a795e585b781.png)

**Comment by jakebailey** [comment]
Okay, I've gotten it in. I won't mark this issue as closed quite yet (as I believe this is the overall tracking issue for "tooltips are too big").

**Comment by CatNofishing** [comment]
☺Thanks for  your reply!

**Comment by CatNofishing** [comment]
To be honest, the readability of parameter prompts is really poor. You can't even quickly distinguish which one is the formal parameter and which is the default parameter. You have to scan such a long function signature one by one, and there is no function document

**Comment by CodeCrazy-ywt** [comment]
@jakebailey @erictraut Here's my suggestions, and it's up to you to take or reject: We can show types in a non-monospaced font(say, Segoe UI) and/or in smaller sizes so that they will occupy less space, and in a different color to make it easier to distinguish them and parameter names apart.

**Comment by judej** [comment]
Moving this issue to discussion as an enhancement request for comments and upvotes.
